io/zenoh-transport/src/unicast/link.rs

```diff
@@ -273,3 +287,61 @@ impl fmt::Debug for TransportLinkUnicastRx {
             .finish()
     }
 }
+
+pub(crate) struct EstablishAck {
+    link: TransportLinkUnicastTx,
+    ack: Option<TransportMessage>,
+}
```

Comment:
It's unclear the need of the `EstablishAck` and of the `ack` functions.

Reply:
`EstablishAck` stores optional open_ack. The `add_link` return value obliges user to unwrap the `EstablishedTransportLinkUnicast` object in an `add_link` method implementation and return `EstablishAck` in case of success, so that the ack could not be forgotten by the developer. The `TransportManager` takes care of using `EstablishAck` object further to send acknowledge in the right moment. If failure happens anywhere, the `EstablishedTransportLinkUnicast` is unwrapped with `fail()` method that drops `EstablishAck`....

Reply:
The input and return arguments around transport initialization methods in the `TransportManager` are designed to put some logical constrains to make initialization procedure more safe: not to forget to send the ack, not to forget to close everything already open in case of failure etc....

Reply:
Ok. Then I'll be more explicit in the naming and in the type. E.g.:
`open_ack: Option<OpenAck>`